pkg/httputil/range_test.go

```diff
@@ -11,41 +12,55 @@ func TestParseRange(t *testing.T) {
 	cases := []struct {
 		Spec          string
 		Length        int
-		ExpectedError bool
+		ExpectedError error
 		ExpectedStart int
 		ExpectedEnd   int
+		ExpectedSize  int
 	}{
-		{"bytes=0-20", 50, false, 0, 20},
-		{"bytes=0-20", 10, false, 0, 9},
-		{"bytes=-20", 50, false, 30, 49},
-		{"bytes=20-", 50, false, 20, 49},
-		{"bytes=-20", 10, true, 0, 0},
-		{"bytes=0-20", 20, false, 0, 19},
-		{"bytes=0-19", 20, false, 0, 19},
-		{"bytes=-0-19", 20, true, 0, 0},
-		{"bytess=0-19", 20, true, 0, 0},
-		{"0-19", 20, true, 0, 0},
-		{"bytes=-", 20, true, 0, 0},
-		{"bytes=0-foo", 20, true, 0, 0},
-		{"bytes=foo-19", 20, true, 0, 0},
-		{"bytes=21-", 20, true, 0, 0},
+		{"bytes=0-20", 50, nil, 0, 20, 21},
+		{"bytes=0-20", 10, nil, 0, 9, 10},
+		{"bytes=-20", 50, nil, 30, 49, 20},
+		{"bytes=20-", 50, nil, 20, 49, 30},
+		{"bytes=-20", 10, nil, 0, 9, 10},
+		{"bytes=0-20", 20, nil, 0, 19, 20},
+		{"bytes=0-19", 20, nil, 0, 19, 20},
+		{"bytes=1-300", 20, nil, 1, 19, 19},
+		{"bytes=-0-19", 20, httputil.ErrBadRange, 0, 0, 0},
+		{"bytess=0-19", 20, httputil.ErrBadRange, 0, 0, 0},
+		{"0-19", 20, httputil.ErrBadRange, 0, 0, 0},
+		{"bytes=-", 20, httputil.ErrBadRange, 0, 0, 0},
+		{"bytes=0-foo", 20, httputil.ErrBadRange, 0, 0, 0},
+		{"bytes=foo-19", 20, httputil.ErrBadRange, 0, 0, 0},
+		{"bytes=20-", 20, httputil.ErrUnsatisfiableRange, 0, 0, 0},
+		{"bytes=21-", 20, httputil.ErrUnsatisfiableRange, 0, 0, 0},
+		{"bytes=19-", 20, nil, 19, 19, 1},
 	}
 
 	for _, c := range cases {
 		t.Run(fmt.Sprintf("%s_length_%d", c.Spec, c.Length), func(t *testing.T) {
 			r, err := httputil.ParseRange(c.Spec, int64(c.Length))
-			if (err != nil) != c.ExpectedError {
-				t.Fatalf("got err=%s, expected error %t", err, c.ExpectedError)
+			if c.ExpectedError != nil && err == nil {
+				t.Fatalf("epxected error: %s, got nil!", c.ExpectedError)
+			}
+			if c.ExpectedError == nil && err != nil {
+				t.Fatalf("unepxected error: %s!", err)
+			}
+			if c.ExpectedError != nil && err != nil && !errors.Is(err, c.ExpectedError) {
+				t.Fatalf("epxected error: %s, got %s!", c.ExpectedError, err)
```

Comment:
You can just use `errors.Is` and use %v while report mismatch as %v will print also the case when err is nil

Reply:
Nice! good call. Fixed.